bson/binary.py

```diff
@@ -400,24 +400,35 @@ def as_uuid(self, uuid_representation: int = UuidRepresentation.STANDARD) -> UUI
     @classmethod
     def from_vector(
         cls: Type[Binary],
-        vector: list[int, float],
-        dtype: BinaryVectorDtype,
-        padding: int = 0,
+        vector: Union[BinaryVector, list[int, float]],
+        dtype: Optional[BinaryVectorDtype] = None,
+        padding: Optional[int] = None,
```

Comment:
Using a typing overload could be a cleaner solution here. Then you could properly convey to the type checker that dtype is required when vector is a list.